diesel/src/associations/mod.rs

```diff
@@ -1,3 +1,113 @@
+//! ```
+//!
+//! Note: Due to language limitations, only two tables can be joined per query. This will change in
+//! the future.
+//!
+//! Typically however, queries are loaded in multiple queries. For most datasets, the reduced
+//! amount of duplicated information sent over the wire saves more than the the extra round trip
+//! costs us. You can load the children for a single parent using the
+//! [`belonging_to`][belonging-to]
+//!
+//! [belonging-to]: prelude/trait.BelongingToDsl.html#tymethod.belonging_to
+//!
+//! ```ignore
+//! let user = try!(users::find(1).first(&connection));
+//! let posts = Post::belonging_to(&user).load(&connection);
+//! ```
+//!
+//! If you're coming from most other ORMs, you'll notice that this design is quite different from
+//! most, where you would have an instance method on the parent, or have the children stored
+//! somewhere on the posts. This design leads to many problems, including N+1 query bugs, and
+//! runtime errors when accessing an association that isn't there.
+//!
+//! In Diesel, data and its associations are considered to be separate. If you want to pass around
+//! a user and all of its posts, that type is `(User, Vec<Post>)`.
+//!
+//! Next lets look at how to load the children for more than one parent record.
+//! [`belonging_to`][belonging-to] can be used to load the data, but we'll also need to group it
+//! with its parents. For this we use an additional method [`grouped_by`][grouped-by]
+//!
+//! [grouped-by]: associations/trait.GroupedBy.html#tymethod.grouped_by
+//!
+//! ```ignore
+//! fn first_twenty_users_and_their_posts(conn: &PgConnection) -> QueryResult<Vec<(User, Vec<Post>)>> {
+//!     let users = try!(users::limit(20).load::<User>(&conn));
+//!     let posts = try!(Post::belonging_to(&users).load::<Post>());
+//!     let grouped_posts = posts.grouped_by(&users);
+//!     users.into_iter().zip(grouped_posts).collect()
+//! }
+//! ```
+//!
+//! [`grouped_by`][grouped-by] takes a `Vec<Child>` and a `Vec<Parent>` and returns a
+//! `Vec<Vec<Child>>` where the index of the children matches the index of the parent the belong
+//! to. Or to put it simply, it returns them in an order ready to be `zip`ed with the parents. You
+//! can do this multiple times. For example, if you wanted to load the comments for all the posts
+//! as well, you could do this:
+//!
+//! ```ignore
+//! let posts = try!(Post::belonging_to(&users).load::<Post>());
+//! let comments = try!(Comment::belonging_to(&posts).load::<Comment>());
+//! let comments = comments.grouped_by(&posts);
+//! let posts_and_comments = posts.into_iter().zip(comments).grouped_by(&users);
+//! users.into_iter().zip(posts_and_comments).collect()
+//! ```
+//!
+//! And that's it. This module will be expanded in the future with more complex joins, and ability
+//! to define "through" associations. However, the goal is to provide simple building blocks which
```

Comment:
Maybe give a short example like "find all the users I follow through the Followers table". (I guess this is what you mean)